Add missing entry in the changelog about the KOKKOS_DEDUCTION_GUIDE macro

tpadioleau opened the pull request.

tpadioleau commented:
Shall I also add an entry in this page of the wiki https://kokkos.org/kokkos-core-wiki/API/core/Macros.html?highlight=macros ?

I think it should belong next to `KOKKOS_FUNCTION`/`KOKKOS_INLINE_FUNCTION` and `KOKKOS_DEFAULTED_FUNCTION` but I don't think they are listed.

masterleinad commented:
> Shall I also add an entry in this page of the wiki [kokkos.org/kokkos-core-wiki/API/core/Macros.html?highlight=macros](https://kokkos.org/kokkos-core-wiki/API/core/Macros.html?highlight=macros) ?

See https://github.com/kokkos/kokkos-core-wiki/pull/578.

masterleinad commented on CHANGELOG.md:
```diff
@@ -83,6 +83,7 @@
 
 ### Bug Fixes
 * Do not return a copy of the input functor in `Experimental::for_each` [\#6910](https://github.com/kokkos/kokkos/pull/6910)
+* Introduce KOKKOS_DEDUCTION_GUIDE macro to allow user-defined deduction guide in device code for clang compiler [\#6954](https://github.com/kokkos/kokkos/pull/6954)
```
```suggestion
* Introduce KOKKOS_DEDUCTION_GUIDE macro to allow for portable deduction guides [\#6954](https://github.com/kokkos/kokkos/pull/6954)
```

tpadioleau replied:
see 88f490276a8ebe94ec2d2a2f747543854157d22f

tpadioleau commented:
@dalg24 Actually I think I got confused by other entries in the changelog 4.4, I don't think this one has ever been in it.

crtrott reviewed (changes requested):
Why under bug fixes? Isn't this more a General Enhancement?

tpadioleau commented:
> Why under bug fixes? Isn't this more a General Enhancement?

I added it in this section because we introduced the macro to fix existing user-defined deduction guides of `TeamThreadMDRange` in device code.

I don't know if we claimed to support in device code though :D
